centreon/www/class/centreonHost.class.php

```diff
@@ -683,14 +673,22 @@ public function getHostPollerId(int $hostId): ?int
      */
     public function replaceMacroInString($hostParam, $string, $antiLoop = null)
     {
+        if (! preg_match('/(?:\$[0-9a-zA-Z_-]+\$)/', $string)) {
+            return $string;
+        }
         if (is_numeric($hostParam)) {
             $hostId = $hostParam;
         } elseif (is_string($hostParam)) {
             $hostId = $this->getHostId($hostParam);
         } else {
             return $string;
         }
-        $query = 'SELECT host_register FROM host WHERE host_id = :hostId LIMIT 1';
+        $query = 'SELECT host_id, nagios_server_id, host_register, host_address, host_name, host_alias
```

Comment:
Please add the alias joined table explicitly in the select (this not obvious for a newbie like me about the db schema).
```suggestion
        $query = 'SELECT host_id, ns.nagios_server_id, host_register, host_address, host_name, host_alias
```